fire/api/model/tidsserier.py

```diff
@@ -424,6 +440,475 @@ def residualkovarians_zz(self) -> List[float]:
+
+        Forudsætter at denne er initialiseret med "forbered_lineær_regression(...)".
+        """
+        self.linreg.solve()
+        return
+
+
+class TidsserieEnsemble:
+    """
+    Gruppér tidsserier og beregn tværgående statistik.
+
+    Tidsserierne i ensemblet grupperes ud fra deres:
+        - tidsserieklasse
+        - tidsseriegruppe
+        - referenceramme
+    """
+
+    def __init__(
+        self,
+        tidsserieklasse: Type,
+        min_antal_punkter: int,
+        tidsseriegruppe: str,
+        referenceramme: str,
+    ):
+        self.tidsserieklasse = tidsserieklasse
+
+        self.tidsserier = {}
+        self.min_antal_punkter = min_antal_punkter
+        self.tidsseriegruppe = tidsseriegruppe
+        self.referenceramme = referenceramme
+
+    def _valider_tidsserie(self, tidsserie: Tidsserie):
+        """Valider en tidsserie inden indsættelse i ensemblet."""
+        if (
+            not isinstance(tidsserie, self.tidsserieklasse)
+            or (len(tidsserie.koordinater) < self.min_antal_punkter)
+            or (tidsserie.tidsseriegruppe != self.tidsseriegruppe)
+            or (tidsserie.referenceramme != self.referenceramme)
+        ):
+            raise ValueError(f"Tidsserie: {tidsserie.navn} kunne ikke valideres")
+
+    def tilføj_tidsserie(self, tidsserie: Tidsserie):
+        """Tilføj en tidsserie til ensemblet."""
+        try:
+            self._valider_tidsserie(tidsserie)
+            print(f"Tilføjer tidsserie {tidsserie.navn}")
+            self.tidsserier.update({tidsserie.navn: tidsserie})
+        except ValueError as e:
+            print(e)
+
+        return
+
+    def beregn_pooled_varians(self):
+        """
+        Beregn den estimerede samlede varians af tidsserieensemblet.
+
+        Den samlede varians beregnes som et vægtet gennemsnit af varianserne
+        af de enkelte tidsserier som indgår i ensemblet, og opdaterer variansen
+        i tidsserierne.
+
+        Kræver at hver tidsserie har en linær regression "linreg"-attribut af typen
+        PolynomieRegression1D, som er blevet løst, ved at få kørt sin "solve-metode."
+        """
+        if not self.tidsserier:
+            raise ValueError(
+                "Ingen tidsserier i ensemble. Kan ikke beregne pooled varians."
+            )
+
+        SSR_sum = 0
+        dof_sum = 0
+
+        # TODO: Kan indsætte try-block her for at fange AttributeError ved tidsserier som ikke har et linreg.SSR eller linreg.dof
+        # Sker hvis tidsserier som indsættes ikke har fået kørt sin "solve"-metode. Alternativ smid check ind i _valider_tidsserie.
+        for ts in self.tidsserier.values():
+            SSR_sum += ts.linreg.SSR
+            dof_sum += ts.linreg.dof
+
+        self.var_pooled = SSR_sum / dof_sum
+
+        # Opdater ensemblets tidsserier med den poolede varians.
+        for ts in self.tidsserier.values():
+            ts.linreg.var_pooled = self.var_pooled
+
+        return
```

Comment:
Returnér noget, eller undlad denne linje, da funktioner altid returnerer til sidst. Et `return` til sidst i funktionen gør det bare eksplicit, men ud over, at det er unødvendigt, så kan det måske virke uklart for en uindviet, at man implicit returnerer `None` ved ikke at skrive nogen eksplicit returværdi.

En anden observation er, at der mangler type hints, hvilket er inkonsistent med resten af koden ovenfor.

Min konklusion er derfor: Tilføj type hints/annotatins, så man blandt andet kan se, at funktionen returnerer None. Fjern det unødvendige `return` udsagn. Skriv gerne i funktionens docstring, at den kun har sideeffekter og derfor ikke returnerer noget, men gemmer resultaterne i instansen.

Reply:
Har rettet dette så alle funktioner der ikke returnerer noget har fået korrekt type hint, samt fjernet overflødige return statements.